src/index.ts

```diff
@@ -134,17 +140,26 @@ export class I2C extends Peripheral {
     super.destroy();
   }
 
-  private getDevice(address: number) {
+  private getDevice(address: number, busNumber?: number) {
     let device = this.devices[address];
 
     if (device === undefined) {
-      device = openSync(getBoardRevision() === VERSION_1_MODEL_B_REV_1 ? 0 : 1);
+      if (busNumber === undefined) {
+        busNumber = this.busNumber;
+      }
+      device = openSync(busNumber);
       this.devices[address] = device;
     }
 
     return device;
   }
 
+  public configure(options?: any): void {
```

Comment:
Unfortunately, because of limitations in how pinMode works and pin contention works, we cannot go with this approach, and the extra bus cannot be made accessible from Johnny-Five until Johnny-Five itself adds support for more than one I2C bus. I'm sorry, but that's a hard limitation.